protocol/carbon/metricdeconstructor/configurableregexdeconstructor.go

```
@@ -0,0 +1,101 @@
+package metricdeconstructor
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/signalfx/golib/datapoint"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+type configurableRegexMetricRule struct {
+	typer
+	Regex                string            `json:"Regex"`
+	MetricName           string            `json:"MetricName"`
+	AdditionalDimensions map[string]string `json:"AdditionalDimensions"`
+	MetricTypeString     string            `json:"MetricType"`
+	MetricType           datapoint.MetricType
+	Compiled             *regexp.Regexp
+	UseCount             int64
+}
+
+type configurableRegexMetricDeconstructor struct {
+	MetricRules                 []*configurableRegexMetricRule `json:"MetricRules"`
+	FallBackDeconstructorName   string                         `json:"FallbackDeconstructor"`
+	FallBackDeconstructorConfig string                         `json:"FallbackDeconstructorConfig"`
+	AdditionalDimensions        map[string]string              `json:"Dimensions"`
+	FallBackDeconstructor       MetricDeconstructor
+}
+
+func regexJSONLoader(config map[string]interface{}) (MetricDeconstructor, error) {
+	var metrics configurableRegexMetricDeconstructor
+	// a wee bit of a hack, but at least we know it's valid json and it makes the config easier
+	jsonString, _ := json.Marshal(config)
+	err := json.Unmarshal([]byte(jsonString), &metrics)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, metricRegex := range metrics.MetricRules {
+		if metricRegex.MetricTypeString != "" {
+			metricType, ok := nameToType[metricRegex.MetricTypeString]
+			if !ok {
+				return nil, fmt.Errorf("cannot parse configured MetricType of %s", metricRegex.MetricTypeString)
+			}
+			metricRegex.MetricType = metricType
+		}
+		metricRegex.Compiled = regexp.MustCompile(metricRegex.Regex)
```

This will panic if the regex isn't valid.  Can you return an error instead?